python_modules/dagster/dagster/_components/__init__.py

```diff
@@ -0,0 +1,32 @@
+from dagster_shared.serdes.objects import LibraryObjectKey as LibraryObjectKey
+
+from dagster._components.component.component import Component as Component
+from dagster._components.component.component_loader import component as component
+from dagster._components.component.component_scaffolder import (
+    DefaultComponentScaffolder as DefaultComponentScaffolder,
+)
+from dagster._components.component_scaffolding import scaffold_component as scaffold_component
+from dagster._components.core.context import ComponentLoadContext as ComponentLoadContext
+from dagster._components.core.load_defs import (
+    build_component_defs as build_component_defs,
+    load_defs as load_defs,
+)
+from dagster._components.resolved.base import Resolvable as Resolvable
+from dagster._components.resolved.context import ResolutionContext as ResolutionContext
+from dagster._components.resolved.core_models import (
+    AssetAttributesModel as AssetAttributesModel,
+    AssetPostProcessorModel as AssetPostProcessorModel,
+    ResolvedAssetSpec as ResolvedAssetSpec,
+)
+from dagster._components.resolved.metadata import ResolvableFieldInfo as ResolvableFieldInfo
+from dagster._components.resolved.model import (
+    Injectable as Injectable,
+    Injected as Injected,
+    Model as Model,
+    Resolver as Resolver,
+)
+from dagster._components.scaffold.scaffold import (
+    Scaffolder as Scaffolder,
+    ScaffolderUnavailableReason as ScaffolderUnavailableReason,
+    ScaffoldRequest as ScaffoldRequest,
+)
```

Comment:
We gotta make sure that the relevant things here are marked with `@preview`. I think the minimal set would be:

- Component
- scaffold_component
- build_component_defs
- load_defs
- Resolvable
- Model
- Scaffolder

Another thing to consider here would be to limit the blast radius of all the `resolved` stuff by tossing that under `dagster.resolved` for now and not exporting that from the top level dagster package. Just a bit spooky to see stuff like:

```
from dagster import Model
```

as that's a pretty general name for somethign that has a pretty specific use case